spinalcordtoolbox/scripts/sct_extract_metric.py

```diff
@@ -358,6 +358,14 @@ def main(fname_data, path_label, method, slices, levels, fname_output, labels_us
     # Load vertebral levels
     if vertebral_levels:
         im_vertebral_labeling = Image(fname_vertebral_labeling).change_orientation("RPI")
+
+        # Get dimensions of input Image and the vertebral labeling file
+        nx, ny, nz = input_im.data.shape
+        nx_vertebral, ny_vertebral, nz_vertebral = im_vertebral_labeling.data.shape
+
+        # Check dimensions consistency between the input Image and the vertebral labeling file
+        if (nx, ny, nz) != (nx_vertebral, ny_vertebral, nz_vertebral):
+            printv('\nERROR: Metric data and vertebral labeling file DO NOT HAVE SAME DIMENSIONS.', 1, type='error')
```

Comment:
shouldn't we be doing this check inside the [API](aggregate_per_slice_or_level) instead? To limit duplication of code in the CLI (here: `sct_extract_metric` and `sct_process_segmentation`). What do you think @joshuacwnewton ?

Reply:
```python
# Rewritten to make it more compact
if input_im.data.shape != im_vertebral_labeling.data.shape:
    printv('\nERROR: Metric data and vertebral labeling file DO NOT HAVE SAME DIMENSIONS.', 1, type='error')
```

Because this is such a simple and self-explanatory check, I think duplication is OK. If we did make an API function (to prevent duplication), I think it might just add clutter to the API, becuase I think users/developers would just write out the check (rather than use the function). (In general, though, for more complex logic, I absolutely agree that avoiding duplication is a helpful mindset to have.) 

Besides that, though, I do agree that we should be doing the check inside the API, but for a different reason. :stuck_out_tongue:

Say a future user calls the relevant API functions (`extract_metric()` and `aggregate_per_slice_or_level()`) in one of their own scripts. If we only do the check in the CLI, then the problem from #3111 would still exist at the API level, I think.

(This is something that `scikit-learn` did quite rigorously when I contributed there. You can see in an example such as [`TSNE`](https://github.com/scikit-learn/scikit-learn/blob/6b4f82433dc2f219dbff7fe8fa42c10b72379be6/sklearn/manifold/_t_sne.py#L681-L735) that they devote a lot of space for validation of function parameters, and I like this.)

Reply:
Thanks for the suggestions from both of you, also sorry about the mistake, I will do the check inside the API.

Reply:
No need to apologize! :slightly_smiling_face:  That's what reviews are for, so we can share different perspectives as a team. :smile: